v2/internal/controller/controllers/generic_controller.go

```diff
@@ -211,6 +219,27 @@ func (gr *GenericReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		"resourceVersion", obj.GetResourceVersion(),
 		"generation", obj.GetGeneration())
 
+	// Ensure the resource is tagged with the operator's namespace.
+	annotations := metaObj.GetAnnotations()
+	reconcilerNamespace := annotations[NamespaceAnnotation]
+	if reconcilerNamespace != gr.PodNamespace && reconcilerNamespace != "" {
+		// We don't want to get into a fight with another operator -
+		// so if we see another operator already has this object leave
+		// it alone. This will do the right thing in the case of two
+		// operators trying to manage the same namespace. It makes
+		// moving objects between namespaces or changing which
+		// operator owns a namespace fiddlier (since you'd need to
+		// remove the annotation) but those operations are likely to
+		// be rare.
+		message := fmt.Sprintf("Operators in %q and %q are both configured to manage this resource", gr.PodNamespace, reconcilerNamespace)
+		gr.Recorder.Event(obj, corev1.EventTypeWarning, "Overlap", message)
+		return ctrl.Result{}, gr.KubeClient.Client.Update(ctx, obj)
```

Comment:
Hmm, I might have been cargo-culting. Having a look now.
Ah, I foolishly thought that since Recorder.Event doesn't return an error it must just store something in the object that we still need to write out, but digging into the implementation it's because it's a fire-and-forget type thing, which makes a lot more sense. I'll remove the update here.